docs/auth/identity-resolver.md

```diff
@@ -402,6 +406,30 @@ async signInResolver({ profile }, ctx) {
 }
 ```
 
+##### Using the `dangerouslyAllowSignInWithoutUserInCatalog` Option
+
+Another way to bypass this requirement is to enable the `dangerouslyAllowSignInWithoutUserInCatalog` option for resolvers. This will allow users to sign in even if their user entity has not been synced into the catalog. This option will issue a Backstage user token using whatever information is available at the resolver level.
+
+For example:
+
+```yaml title="Within the provider configuration"
+auth:
+  providers:
+    github:
+      development:
+        ...
+        signIn:
+          resolvers:
+            - resolver: emailLocalPartMatchingUserEntityName
+              dangerouslyAllowSignInWithoutUserInCatalog: true
+```
+
+:::warning
+Enabling this option in production poses security risks.
+:::
+
+This option grants access to unexpected users who haven’t been onboarded into Backstage. It also breaks integration with permission frameworks like the RBAC plugin, since there is no user entity to associate with the signed-in user, potentially leading to unauthorized access to resources.
```

Comment:
Hello! 👋 

Let me know if I am misunderstanding, but enabling this will still mean that users are authenticated right? they still go through the authentication that is in place, we just don't additionally check if they exist in the catalog (which is what I gathered from the sig) 

I think as we don't assign them anything as a fallback (which could be cool thing to do maybe later on maybe as config) we can simply say that the permissions framework or just permissions will not be applied to these users at all as they technically do not exist in the catalog.

If the first statement is true though and I understood correctly I'd definitely mention that part somewhere 😄 

Also this is really cool and I think definitely something we'd use!

Thanks!

Reply:
Hi! Thank you so much for your feedback, you are right and users will still be authenticated. I've updated the docs to make this clearer :)

I've also added more detail on how this interacts with permissions (tested with the RBAC plugin), let me know how it looks!

Reply:
Nice! LGTM! 👏